odooconnector_base/models/product.py

```diff
@@ -0,0 +1,268 @@
+# -*- coding: utf-8 -*-
+# © 2015 Malte Jacobi (maljac @ github)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import logging
+
+from openerp import models, fields
+from openerp.addons.connector.unit.mapper import (mapping, ExportMapper)
+
+from ..unit.import_synchronizer import (OdooImporter, DirectBatchImporter,
+                                        TranslationImporter)
+from ..unit.export_synchronizer import (OdooExporter, TranslationExporter)
+from ..unit.mapper import (OdooImportMapper, OdooImportMapChild,
+                           OdooExportMapChild)
+from ..unit.backend_adapter import OdooAdapter
+from ..backend import oc_odoo
+
+
+_logger = logging.getLogger(__name__)
+
+
+class OdooConnectorProductTemplate(models.Model):
+    _name = 'odooconnector.product.product'
+    _inherit = 'odooconnector.binding'
+    _inherits = {'product.product': 'openerp_id'}
+    _description = 'Odoo Connector Product'
+
+    openerp_id = fields.Many2one(
+        comodel_name='product.product',
+        string='Product',
+        required=True,
+        ondelete='restrict'
+    )
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    oc_bind_ids = fields.One2many(
+        comodel_name='odooconnector.product.product',
+        inverse_name='openerp_id',
+        string='Odoo Connector Bindings'
+    )
+
+
+@oc_odoo
+class ProductBatchImporter(DirectBatchImporter):
+    _model_name = ['odooconnector.product.product']
+
+
+@oc_odoo
+class ProductTranslationImporter(TranslationImporter):
+    _model_name = ['odooconnector.product.product']
+
+
+@oc_odoo
+class ProductImportMapper(OdooImportMapper):
+    _model_name = ['odooconnector.product.product']
+    _map_child_class = OdooImportMapChild
+
+    direct = [('name', 'name'), ('name_template', 'name_template'),
+              ('type', 'type'),
+              ('purchase_ok', 'purchase_ok'), ('sale_ok', 'sale_ok'),
+              ('lst_price', 'lst_price'), ('standard_price', 'standard_price'),
+              ('ean13', 'ean13'), ('default_code', 'default_code'),
+              ('description', 'description')]
```

Comment:
I understand your objections and in some scenarios they are true. In the scenarios that I implemented so far it was necessary to sync all those fields.

What I did in a private fork (which I will open source in Q1/16) is the configuration option (in the frontend!) to select fields that should be excluded from syncing.

What we could discuss is:
1. should we sync max fields and exclude fields that should not be synced or 
2. should we sync min viable number of fields and define additional fields per use case

*Pro / Cons (1)*

Pro: Some more complex logic than `field1=field1` can be used without dev
Con: "Exclude from Sync" list might be long if you need only min. viable fields 
-